frontend/components/common/InstantSearchProvider.tsx

```diff
@@ -230,7 +230,10 @@ export function InstantSearchProvider({
 
             return {
                q: String(q || ''),
-               p: typeof p === 'string' ? parseInt(p) : undefined,
+               p:
+                  typeof p === 'string' && parseInt(p) >= 0
+                     ? parseInt(p)
+                     : undefined,
```

Comment:
Looking at the code it seems that `p` can by of type `any`.

However, `parseInt()` will return `NaN` if it can't parse the integer https://developer.mozilla.org/en-US/docs/web/javascript/reference/global_objects/parseint

So we can extract the parsing to a line above and then simply check the negative value in this line. This will make the line less bloated and not have to run `parseInt` twice.

So something like 

```

const pageParam = parseInt(p)

return {
   q: String(q || ''),

   p: pageParam >= 0 ? pageParam : undefined

```

[Typescript Example](https://www.typescriptlang.org/play?#code/DYUwLgBADgXBCGA7AngbgFCkleAnAziACYD6UEAvNHoQJKJgAUUAlOgMYD2i+noAdME4BzRgAMAgimrCQEAAp54AWwgASAN44CIek1YBfMW0zgZcqgCIAjAAZLp7DWJl4sytR17mbkGy48fCCCIuLynPgAlmCRAG5yOO6KuCrqWs7eiX5GJlgQiCDC8DHxru5WALQALA552oSkBUUlIGUWnnQMjE3Fca1Z-ty8AkKiYgByhb3x5gpKqpr1ul09LW0sOegcQ0EhY1LIswBmkYjwwGlLpOQAfFS2EAD8HS7kcACuiEQgJwVEmwFhsFRmEItE+sdTudLs5rr4IHcIA9nlc2hAPl8fqdiACdiNQhMpi1IWcLotYSRVn00YjkS9GkTqVl0RBPt9fjjjNtAvjRON4OMEfcICwgA)